keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,928 @@
+- Feature is enabled by default.
+- Address all issues reported by users.
+- Decided whether we introduce a configurable reason for the `SuccessCriteriaMet` condition.
+Please see the [Possibility for the configurable reason for the "SuccessCriteriaMet" condition](#possibility-for-the-configurable-reason-for-the-successcriteriamet-condition). 
+
+#### GA
+
+- No negative feedback.
+
+### Upgrade / Downgrade Strategy
+
+- Upgrade
+  - If the feature gate is enabled, `JobSuccessPolicy` are allowed to use only.
+  - If the feature gate is enabled without `JobSuccessPolicy`, 
+  the default values will be applied to a job object.
+  - Even if the feature gate is enabled, the Job controller doesn't update
+  `.status.conditions` in already finished jobs.
+- Downgrade
+  - Previously configured values will be ignored, and the job will be marked
+  as completed only when all indexes succeed.
+  - the Job controller doesn't update `.status.conditions` in already finished jobs.
+
+### Version Skew Strategy
+
+The apiserver's version should be consistent with the kube-controller-manager version,
+or this feature will not work.
+
+This feature is limited to control plane.
+
+Note that, the kube-apiserver can be in the N+1 skew version relative to the
+kube-controller-manager as described [here](https://kubernetes.io/releases/version-skew-policy/#kube-controller-manager-kube-scheduler-and-cloud-controller-manager).
+If it's enabled, jobs with SuccessPolicy set will have it respected. Otherwise, it will be ignored by the job controller.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: JobSuccessPolicy
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+No, the default behavior of a job and cronJob stays the same.
+The newly added field is optional and has to be explicitly set by the user to use this new feature.
+
+Regarding the CronJob, please see more details in [#The CronJob concurrentPolicy is not affected by JobSuccessPolicy](#the-cronjob-concurrentpolicy-is-not-affected-by-jobsuccesspolicy).
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, we can disable the `JobSuccessPolicy` feature gate.
+When the feature is disabled, the job controller stop evaluating the successPolicy even if
+the `.spec.successPolicy` is set.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The Job controller considers the `.spec.successPolicy` when it updates `.status.conditions`
+only for running Jobs and don't update `.status.conditions` for already finished jobs.
+
+###### Are there any tests for feature enablement/disablement?
+
+TODO
```

Comment:
The comment from the template is relevant for this case:
```

Additionally, for features that are introducing a new API field, unit tests that
are exercising the `switch` of feature gate itself (what happens if I disable a
feature gate after having objects written with the new field) are also critical.
You can take a look at one potential example of such test in:
https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
-->
```

Can you add here that for alpha you will at least add the appropriate strategy test here for newly added API fields?

Reply:
That makes sense.

Reply:
Done.